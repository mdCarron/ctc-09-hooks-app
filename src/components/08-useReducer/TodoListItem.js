import React from "react";

const TodoListItem = ({ todo, i, handleToggle, handleDelete }) => {
  return (
    <li
      key={todo.id}
      className="list-group-item d-flex justify-content-between align-items-center"
    >
      <p
        className={`${todo.done && "completed"}`}
        onClick={() => handleToggle(todo.id)}
      >
        {i + 1}. {todo.desc}
      </p>
      <button
        className="btn btn-outline-danger"
        onClick={() => handleDelete(todo.id)}
      >
        Borrar
      </button>
    </li>
  );
};

export default TodoListItem;
