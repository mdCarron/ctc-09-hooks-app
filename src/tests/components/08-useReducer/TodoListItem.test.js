import React from "react";
import { shallow } from "enzyme";
import { demoTodos } from "../../fixtures/demoTodos";
import TodoListItem from "../../../components/08-useReducer/TodoListItem";

describe("<TodoListItem />", () => {
  const handleToggle = jest.fn();
  const handleDelete = jest.fn();

  const wrapper = shallow(
    <TodoListItem
      todo={demoTodos[0]}
      i={0}
      handleToggle={handleToggle}
      handleDelete={handleDelete}
    />
  );

  test("should match snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("should call handleDelete function", () => {
    wrapper.find("button").simulate("click");
    expect(handleDelete).toHaveBeenCalledWith(demoTodos[0].id);
  });

  test("should call handleToggle function", () => {
    wrapper.find("p").simulate("click");
    expect(handleToggle).toHaveBeenCalledWith(demoTodos[0].id);
  });

  test("should render text with variables", () => {
    const p = wrapper.find("p");
    expect(p.text().trim()).toBe(`1. ${demoTodos[0].desc}`);
  });

  test("should if todo.done = true, class = complete", () => {
    const todo = demoTodos[0];
    todo.done = true;

    const wrapper = shallow(<TodoListItem todo={todo} />);

    expect(wrapper.find("p").hasClass("completed")).toBe(true);
  });
});
