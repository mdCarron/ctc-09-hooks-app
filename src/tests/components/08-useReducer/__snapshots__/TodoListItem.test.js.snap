// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TodoListItem /> should match snapshot 1`] = `
<li
  className="list-group-item d-flex justify-content-between align-items-center"
  key="1"
>
  <p
    className="false"
    onClick={[Function]}
  >
    1
    . 
    Aprender React
  </p>
  <button
    className="btn btn-outline-danger"
    onClick={[Function]}
  >
    Borrar
  </button>
</li>
`;
